Tests/L0/Maven/_suite.ts

```diff
@@ -44,42 +44,55 @@ describe('Maven Suite', function() {
     it('Maven / PMD: Executes PMD goals if PMD is enabled', (done) => {
         // Arrange
 
-        var testStgDir:string = path.join(__dirname, '_temp');
-        var testSrcDir:string = path.join(__dirname, 'data');
-        tl.rmRF(testStgDir);
-        tl.mkdirP(testStgDir);
+        var agentSrcDir:string = path.join(__dirname, 'data');
+        var agentStgDir:string = path.join(__dirname, '_temp');
+        var codeAnalysisStgDir:string = path.join(agentStgDir, '.codeanalysis'); // overall directory for all tools
+        var pmdStgDir:string = path.join(codeAnalysisStgDir, '.pmd'); // PMD subdir is used for artifact staging
 
-        tl.mkdirP(path.join(testStgDir, '.pmd')); // Manually create the .pmd subfolder for test purposes
 
-        // Add test file(s) to the response file so that tl.exist() and tl.checkPath() calls return correctly
-        var testXmlFilePath = path.join(testSrcDir, 'target', 'pmd.xml');
-        var testHtmlFilePath = path.join(testSrcDir, 'target', 'site', 'pmd.html');
+        tl.rmRF(agentStgDir);
+
+        // Create folders for test
+        tl.mkdirP(agentStgDir);
+        tl.mkdirP(codeAnalysisStgDir);
+        tl.mkdirP(pmdStgDir);
+
+        // Add set up the response file so that task library filesystem calls return correctly
+        // e.g. tl.exist(), tl.checkpath(), tl.rmRF(), tl.mkdirP()
+        var testXmlFilePath = path.join(agentSrcDir, 'target', 'pmd.xml');
+        var testHtmlFilePath = path.join(agentSrcDir, 'target', 'site', 'pmd.html');
         var responseJsonFilePath:string = path.join(__dirname, 'mavenPmdGood.json');
         var responseJsonContent = JSON.parse(fs.readFileSync(responseJsonFilePath, 'utf-8'));
 
         responseJsonContent.exist = responseJsonContent.exist || {}; // create empty object only if it did not already exist
+        responseJsonContent.exist[agentStgDir] = true;
+        responseJsonContent.exist[codeAnalysisStgDir] = true;
+        responseJsonContent.exist[pmdStgDir] = true;
         responseJsonContent.exist[testXmlFilePath] = true;
         responseJsonContent.exist[testHtmlFilePath] = true;
 
         responseJsonContent.checkPath = responseJsonContent.checkPath || {};
+        responseJsonContent.checkPath[agentStgDir] = true;
+        responseJsonContent.checkPath[codeAnalysisStgDir] = true;
+        responseJsonContent.checkPath[pmdStgDir] = true;
```

Comment:
The volume of these mock settings started to become cumbersome, and have beem refactored out into a different method in a commit to be reviewed in the future.